src/lib/Libattr/attr_fn_resc.c

```diff
@@ -453,72 +454,61 @@ free_resc(attribute *pattr)
 
 /**
  * @brief
- * 	find_resc_def - find the resource_def structure for a resource with
- *	a given name
+ * 	 create the search index for resource deinitions
  *
  * @param[in] rscdf - address of array of resource_def structs
- * @param[in] name - name of resource
  * @param[in] limit - number of members in resource_def array
  *
- * @return	pointer to structure
- * @retval	pointer to resource_def structure	Success
- * @retval	NULL					Error
+ * @return	error code
+ * @retval	0  - Success
+ * @retval	-1 - Failure
  *
  */
-
-resource_def *
-find_resc_def(resource_def *rscdf, char *name, int limit)
+int
+cr_rescdef_idx(resource_def *resc_def, int limit)
 {
-	if (rscdf == NULL || name == NULL)
-		return NULL;
+	int i;
+
+	if (!resc_def)
+		return -1;
 
-	while (limit--) {
-		if (strcasecmp(rscdf->rs_name, name) == 0)
-			return (rscdf);
-		rscdf = rscdf->rs_next;
+	/* create the attribute index */
+	if ((resc_attrdef_idx = pbs_idx_create(PBS_IDX_CASE_CMP, 0)) == NULL)
+		return -1;
+
+	/* add all attributes to the tree with key as the attr name */
+	for (i = 0; i < limit; i++) {
+		if (strcmp(resc_def->rs_name, RESC_NOOP_DEF) != 0) {
+			if (pbs_idx_insert(resc_attrdef_idx, resc_def->rs_name, resc_def) != PBS_IDX_RET_OK)
+				return -1;
+		}
+		resc_def++;
 	}
-	return NULL;
+	return 0;
 }
 
 /**
  * @brief
- *	Determines if a resource is a PBS built-in or user custom
+ * 	find the resource_def structure for a resource with a given name
  *
  * @param[in] rscdf - address of array of resource_def structs
+ * @param[in] name - name of resource
+ * @param[in] limit - number of members in resource_def array
  *
- * @return	int
- * @retval 	1 	if built-in
- * @retval 	0 	if custom
- * @retval 	-1 	on error
+ * @return	pointer to structure
+ * @retval	pointer to resource_def structure - Success
+ * @retval	NULL - Error
  *
  */
-int
-is_builtin(resource_def *rscdef)
+resource_def *
+find_resc_def(resource_def *resc_def, char *name, int limit)
```

Comment:
yes we don't need, but there is also no benefit of removing it. I left it in case we ever need this in future. In the case of find_attr() i removed it since i had to change the definition anyway (to include the idx parameter), however find_resc_def() defn does not require any change. I can remove it, of course, if i find some time

Reply:
Update: found time and removed it :)